corehq/form_processor/document_stores.py

```diff
@@ -46,7 +49,7 @@ def get_document(self, doc_id):
 
     def iter_document_ids(self, last_id=None):
         # todo: support last_id
-        return iter(self.case_accessors.get_case_ids_in_domain())
+        return iter(self.case_accessors.get_case_ids_in_domain(type=self.case_type))
```

Comment:
not a comment on this PR but getting ALL ids for ICDS / eNikshay doesn't seem like a good idea

Reply:
Indeed it does not. which is why i created the async_rebuild_table management command to get around this. This should at least unblock most rebuilds on cases. only ones I would be concerened about at the moment is adherence for enikshay and person for icds

Reply:
could do something like this too: https://github.com/dimagi/commcare-hq/blob/14b16a17c964ca52e1b44937286385833b219e8f/corehq/ex-submodules/pillowtop/reindexer/change_providers/case.py#L30

Reply:
Actually I was considering changing the databases definition to do that exact same thing. Weren't we planning on doing all the partitioning in python instead of plproxy? Could at least create a different implementation to return an iterator instead of a list